### PR TITLE
OpenDUNE: SDL game

### PR DESCRIPTION
**Pull request by lonemadmax**
Cleanup and continue the work by @theclue in #3275.

Known issues:
- The keyboard now works, but it assumes a QWERTY US layout (shift-8 gives '*' instead of the '(' that my keyboard has, for example). The original game under DOS also does this, so I guess it's OK, given the project objective. I don't know how bad it may be for AZERTY, DVORAK, non-latin keyboards...
- The icon is still theclue's draft one. There are png, ico and icns in the project files with another one, but my skills for icon creation are null, and I think theclue's one look more Haikuish in its simplicity anyway.
- I see a bunch of "instrument not found" messages during music play.

### CODE REVIEW

File: games-strategy/opendune/patches/opendune-0.9.patchset [hunk 1]
```diff
@@ -0,0 +1,869 @@
...
+ 	if (f == NULL) {
+ 		/* current directory */
+ 		f = fopen("opendune.ini", "rb");
+-- 
+2.19.1
+
+
+From a928be7dbb7cfb4d6c8c168b269c6decf38f6fa9 Mon Sep 17 00:00:00 2001
+From: Gabriele Baldassarre <gabriele@gabrielebaldassarre.com>
+Date: Fri, 26 Oct 2018 01:50:10 +0200
```
Comment (korli) [inline games-strategy/opendune/patches/opendune-0.9.patchset:193]: it would be nice to squash your own commits.

File: games-strategy/opendune/opendune-0.9.recipe [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	devel:libSDL_image$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:git
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:sdl_config
```
Comment (korli) [inline games-strategy/opendune/opendune-0.9.recipe:55]: Should be cmd:sdl_config$secondaryArchSuffix

File: games-strategy/opendune/opendune-0.9.recipe [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libSDL$secondaryArchSuffix
+	devel:libSDL_image$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:git
```
Comment (korli) [inline games-strategy/opendune/opendune-0.9.recipe:50]: cmd:git comes after cmd:gcc

File: games-strategy/opendune/opendune-0.9.recipe [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+	cmd:sdl_config
+	cmd:sed
+	cmd:which
+	"
+
+BUILD()
+{
+	_dune2_data_dir=$(finddir B_USER_NONPACKAGED_DATA_DIRECTORY)/opendune
+
+	export CFLAGS="-mmmx -DDUNE_DATA_DIR='\\\"${_dune2_data_dir}\\\"' $CFLAGS"
```
Comment (korli) [inline games-strategy/opendune/opendune-0.9.recipe:64]:
```suggestion
	export CFLAGS="-mmmx -DDUNE_DATA_DIR='\\\"${_dune2_data_dir}\\\"'"
```

File: games-strategy/opendune/opendune-0.9.recipe [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	export LDFLAGS="-lstdc++ $LDFLAGS"
```
Comment (korli) [inline games-strategy/opendune/opendune-0.9.recipe:65]:
```suggestion
	export LDFLAGS="-lstdc++"
```

File: games-strategy/opendune/opendune-0.9.recipe [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+INSTALL()
+{
+	mkdir -p $dataDir/opendune/
+
+	make install DO_NOT_INSTALL_DOCS=1 DO_NOT_INSTALL_LICENSE=1
+
+	sed  -e "s|the desired location|$(finddir B_USER_SETTINGS_DIRECTORY)\/opendune|g" $sourceDir/bin/opendune.ini.sample > $sourceDir/bin/opendune.ini.tmp && mv $sourceDir/bin/opendune.ini.tmp $sourceDir/bin/opendune.ini.sample
+	sed  -e "s|;savedir=\/home\/user\/.config\/opendune|savedir=$(finddir B_USER_SETTINGS_DIRECTORY)\/opendune/savegames|g" $sourceDir/bin/opendune.ini.sample > $sourceDir/bin/opendune.ini.tmp && mv $sourceDir/bin/opendune.ini.tmp $sourceDir/bin/opendune.ini.sample
```
Comment (korli) [inline games-strategy/opendune/opendune-0.9.recipe:85]:
```suggestion
	sed  -e "s|the desired location|$(finddir B_USER_SETTINGS_DIRECTORY)\/opendune|g" \
		$sourceDir/bin/opendune.ini.sample > $sourceDir/bin/opendune.ini.tmp \
		&& mv $sourceDir/bin/opendune.ini.tmp $sourceDir/bin/opendune.ini.sample
	sed  -e "s|;savedir=\/home\/user\/.config\/opendune|savedir=$(finddir B_USER_SETTINGS_DIRECTORY)\/opendune/savegames|g" \
		$sourceDir/bin/opendune.ini.sample > $sourceDir/bin/opendune.ini.tmp \
		&& mv $sourceDir/bin/opendune.ini.tmp $sourceDir/bin/opendune.ini.sample
```

File: games-strategy/opendune/opendune-0.9.recipe (hunk 2 above)
Comment (Begasus) [reply games-strategy/opendune/opendune-0.9.recipe:55]: This shouldn't be needed as it's provided by:
```
devel:libSDL$secondaryArchSuffix
```

File: games-strategy/opendune/opendune-0.9.recipe [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libSDL$secondaryArchSuffix
+	lib:libSDL_image$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libSDL$secondaryArchSuffix
+	devel:libSDL_image$secondaryArchSuffix
```
Comment (Begasus) [inline games-strategy/opendune/opendune-0.9.recipe:46]: While we are at it, does it build with SDL1.2 then maybe check requirements here: https://github.com/haikuports/haikuports/issues/3878

File: games-strategy/opendune/opendune-0.9.recipe (hunk 2 above)
Comment (korli) [reply games-strategy/opendune/opendune-0.9.recipe:55]: Indeed

**Comment by Begasus** [comment]
Merged, keep track on the buildbot if the build succeeds and in case of issues, thanks!

**Comment by Begasus** [comment]
Build failed on x86_gcc2, probably as mentioned in:
https://github.com/haikuports/haikuports/pull/3275#issuecomment-584095359

https://build.haiku-os.org/buildmaster/master/x86_gcc2/logviewer.html?buildruns/717/builds/1197.log
